shared/js/background/classes/ad-click-attribution-policy.js

```diff
@@ -173,4 +198,50 @@ export class AdClick {
         if (tab.site.baseDomain !== this.adBaseDomain) return false
         return this.hasNotExpired()
     }
+
+}
+
+export class AdClickDNR {
+    constructor (tabId, allowlist) {
+    this.allowlist = allowlist
+    this.tabId = tabId
+    this.initiatorDomain = null
+    this.rule  = generateDNRRule({
+            "id" : tabId,
+            "priority": CEILING_PRIORITY,
+            "actionType": "allow",
+            "requestDomains": allowlist.reduce((lst, entry) => { lst.push(entry.host); return lst}, []),
```

Comment:
Doesn't this do the same thing:

```suggestion
            "requestDomains": allowlist.map((entry) => entry.host),
```